File: core/src/main/java/fr/sncf/osrd/stdcm/OccupancyBlock.java
package fr.sncf.osrd.stdcm;

/** The given element is unavailable from timeStart until timeEnd,
 * in the space between distanceStart and distanceEnd.
 * Distances are relative to the start of the route. */
public record OccupancyBlock(
        double timeStart,
        double timeEnd,
        double distanceStart,
        double distanceEnd
){}

File: core/src/main/kotlin/fr/sncf/osrd/stdcm/OccupancyBlock.kt
package fr.sncf.osrd.stdcm

/** The given element is unavailable from timeStart until timeEnd,
 * in the space between distanceStart and distanceEnd.
 * Distances are relative to the start of the route.  */
@JvmRecord
data class OccupancyBlock(
    @JvmField val timeStart: Double,
    @JvmField val timeEnd: Double,
    @JvmField val distanceStart: Double,
    @JvmField val distanceEnd: Double
)
